portfolio/src/main/webapp/script.js

```diff
@@ -65,3 +65,186 @@ function addRandomFunFact() {
+  fetch('/data').then(response => response.json()).then(comments => {
+    const commentsListElement = document.getElementById('comments-list');
+    commentsListElement.innerHTML = '';
+
+    const commentsPerPageElement = document.getElementById('num-comments-per-page');
+    const commentsPerPage = commentsPerPageElement.options[commentsPerPageElement.selectedIndex].value;
+
+    const currentPage = document.querySelector('.page-item.active');
+    var currentPageNumber = currentPage.firstElementChild.innerText;
+    if (createNewPagination) {
+      currentPageNumber = createPaginationBar(Math.ceil(comments.length / commentsPerPage), currentPageNumber);
+    }
+
+    const startIndex = (currentPageNumber - 1) * commentsPerPage;
+    const endIndex = Math.min(Number(commentsPerPage) + Number(startIndex), comments.length);
+    for (var i = startIndex; i < endIndex; i++) {
+      commentsListElement.appendChild(createCommentElement(comments[i]));
+    }
+  });
+}
+
+/**
+ * Creates an <li> element containing a comment, including the username, content, time, and delete button.
+ * @param {!Comment} comment The Comment object from which a <li> element is created.
+ * @return {!Element<li>} The comment element created.
+ */
+function createCommentElement(comment) {
+  const commentElement = document.createElement('li');
+  commentElement.className = 'comment';
+
+  const commentContainer = document.createElement('div');
+
+  const nameElement = document.createElement('strong');
+  nameElement.className = 'text-success';
+  nameElement.innerText = '@' + comment.name + ' - ' + comment.mood;
+  const timeElement = document.createElement('span');
+  timeElement.className = 'pull-right text-muted';  // Add Bootstrap classes to style the timeElement.
+  timeElement.innerText = convertToDateTime(comment.timestamp);
+  const contentElement = document.createElement('p');
+  contentElement.innerText = comment.content;
+  
+  commentContainer.appendChild(timeElement);
+  commentContainer.appendChild(nameElement);
+  commentContainer.appendChild(contentElement);
+
+  const deleteButton = document.createElement('button');
+  deleteButton.innerText = 'Delete';
+  deleteButton.addEventListener('click', () => {
+    deleteComment(comment);
+    commentElement.remove();  // Remove the comment from the DOM.
+  });
+
+  commentElement.appendChild(commentContainer);
+  commentElement.appendChild(deleteButton);
+  return commentElement;
+}
+
+/**
+ * Converts a timestamp in milliseconds to a formatted date/time string.
+ * @param {string} timestamp The timestamp to be converted. 
+ * @return {string} Date with format dd/mm/yyyy if the given time is not today; 
+ *     otherwise, number of hours before the current time. 
+ */
+function convertToDateTime(timestamp) {
+  var date = new Date(timestamp);
+  var dateFormatted = date.toLocaleDateString(); 
+  var today = new Date();
+  if (dateFormatted == today.toLocaleDateString()) {
```

Comment:
You might want to add a comment above this `if` case to say what exactly it's doing.

Reply:
Done